src/averageAges.js

```diff
@@ -59,6 +88,37 @@ function calculateAverageAgeDiff(people, onlyWithSon) {
   // 2. keep people who have mothers in the array
   // 3. calculate the difference child.born - mother.born
   // 4. return the average value
+  const filterChildren = people.filter((person) => {
+    if (person.mother === null) {
+      return false;
+    }
+
+    const motherExist = people.some((mother) => mother.name === person.mother);
+
+    if (onlyWithSon) {
+      return motherExist && person.sex === 'm';
+    }
+
+    return motherExist;
+  });
+
+  const ageDifference = filterChildren.map((child) => {
+    const mother = people.find((person) => {
+      return person.name === child.mother;
+    });
+
+    return child.born - mother.born;
+  });
+
+  return calculateAverageAgeDifference(ageDifference);
+}
+
+function calculateAverageAgeDifference(ageDiff) {
+  const average = ageDiff.reduce((prev, age) => {
+    return prev + age;
+  }, 0);
+
+  return average / ageDiff.length;
 }
```

Comment:
Make this more generic
```suggestion
function calculateAverage(numbers) {
  const sum = numbers.reduce((prev, next) => {
    return prev + next;
  }, 0);

  return sum / numbers.length;
}
```